pootle/apps/pootle_app/models/directory.py

```diff
@@ -33,13 +34,22 @@ def projects(self):
         return self.get(pootle_path='/projects/')
 
 
+def validate_no_slashes(value):
+    if '/' in value:
+        raise ValidationError('Directory name cannot contain "/" characters')
+
+    if '\\' in value:
+        raise ValidationError('Directory name cannot contain "\\" characters')
+
+
 class Directory(models.Model, CachedTreeItem):
 
     # any changes to the `name` field may require updating the schema
     # see migration 0005_case_sensitive_schema.py
-    name = models.CharField(max_length=255, null=False)
+    name = models.CharField(max_length=255, null=False, blank=True,
+                            validators=[validate_no_slashes])
     parent = models.ForeignKey('Directory', related_name='child_dirs',
-                               null=True, db_index=True)
+                               null=True, blank=True, db_index=True)
```

Comment:
Is there a remote chance to create a root directory (i.e. a directory with a blank parent) from a form? I don't believe so... (Note how [`blank` only affects form validation](https://docs.djangoproject.com/en/1.8/ref/models/fields/#blank)).